wspr_rx.sh

```diff
@@ -9,24 +9,20 @@ then
 fi
 echo "WSPR Decoder by M0WUT: Callsign = $1, Locator = $2, Frequency = $3"
 
+cd /tmp
 start_time=$(date +"%y%m%d %H%M")
-echo "Beginning recording"
-sudo arecord -f S16_LE -r 48000 --duration=114 -q -t wav -D hw:0,0 -c 2 /mnt/ramdisk/raw.wav
-echo "Recording complete"
-sox /mnt/ramdisk/raw.wav -c 1 -r 12000 output.wav
-echo "Decimation complete"
-~/wspr_rx/wsprd -wf "$3" test.wav >/dev/null 2>&1
-sed -i 's/tput   //g' ALL_WSPR.TXT
-sed -i "s/^....../$start_time/g" ALL_WSPR.TXT
+
+echo "Recording..."
+arecord -f S16_LE -r 48000 --duration=114 -q -t wav -D hw:0,0 -c 2 | sox -t wav -c 1 -r 12000 - - | wsprd -wf "$3" /dev/stdin >/dev/null 2>&1
```

Comment:
Can `wsprd` change the output file? If so we can have it write to `/dev/stdout` straight away and pipe it to `sed`. Even more pipes!

Reply:
Not sure, the c source code that it is running is here if you are interested / think that may be a better way to do it. It's a bit grim as that guy loves his variable names to be... concise, shall we say: https://github.com/k9an/old_wsprcan Line of interest is 422 in wsprd.c

Reply:
By not sure, I mean. No, it's hard coded but you may be able to edit the source to allow it, not sure if fopen works with stdout

Reply:
Ah, don't worry then. `fopen` will work with any file, but if it's hard-coded probably easier just to open the file.